File: 5.implementation/NT/redsun-service/src/main/java/com/redsun/service/service/ActivityPricesService.java
package com.redsun.service.service;

import com.redsun.service.entities.ActivityPrices;
import com.redsun.service.entities.Result;

/**
 * ActivityPrices Service interface
 */
public interface ActivityPricesService {
	
    // Insert.
	Result insert(final ActivityPrices activityPrices) ;

    // Update.
	Result update(final ActivityPrices activityPrices) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listActivityPricesForPageAndFilter(final int itemsPerPage, final int pageNo, final ActivityPrices activityPrices);

}

File: 5.implementation/NXP/redsun/src/main/webapp/static/js/services/ActivityPricesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('activityPriceService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(activityPrice) {
			var urlServer = $rootScope.makePostURL("/activityprice/insert");
			return $http.post(urlServer, activityPrice);
		}
		
		// Update.
		, update: function(activityPriceId, activityPrice) {
			var urlServer = $rootScope.makePostURL("/activityprice/update/" + activityPriceId);
			return $http.put(urlServer, activityPrice);
		}
		
		// Delete.
		, delete: function(activityPriceId) {
			var urlServer = $rootScope.makePostURL("/activityprice/delete/" + activityPriceId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(activityPriceId) {
			var url = $rootScope.makeGetURL("/activityprice/getbyid/" + activityPriceId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, activityPrice) {
			var url = $rootScope.makeGetURL("/activityprice/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: activityPrice });
		}
		
	}// return.
});